resources/benchmark-runner.mjs

```diff
@@ -163,6 +163,166 @@ const WarmupSuite = {
     ],
 };
 
+class TimerCallbacks {
+    constructor(firstCallback, secondCallback) {
+        this._firstCallback = firstCallback;
+        this._secondCallback = secondCallback;
+        this._setTimeoutCallback = this._setTimeout.bind(this);
+    }
+
+    run() {
+        setTimeout(this._setTimeoutCallback, 0);
+    }
+
+    _setTimeout() {
+        this._firstCallback();
+        setTimeout(this._secondCallback, 0);
+    }
+}
+
+class RAFBracketedCallbacks {
+    constructor(firstCallback, secondCallback) {
+        this._firstCallback = firstCallback;
+        this._secondCallback = secondCallback;
+        this._setTimeoutCallback = this._setTimeout.bind(this);
+    }
+
+    run() {
+        requestAnimationFrame(this._firstCallback);
+        requestAnimationFrame(this._setTimeoutCallback);
+    }
+
+    _setTimeout() {
+        setTimeout(this._secondCallback, 0);
+    }
+}
+
+class MeasureStep {
+    constructor(runner, suite, test, page, frame) {
+        this._runner = runner;
+        this.suite = suite;
+        this.test = test;
+        this._page = page;
+        this._frame = frame;
+
+        this._unusedHeight = undefined;
+
+        this._wasRun = false;
+
+        this._asyncMeasurePromise = new Promise((resolve) => {
+            this._asyncDoneCallback = resolve;
+        });
+        this._rAFMeasurement = new RAFBracketedCallbacks(this._measureRafStart.bind(this), this._measureRafEnd.bind(this));
```

Comment:
Something here is rather confusing. Using RAFBracketedCallbacks in two places with different callbacks.
Why do we need this kind of setup?